src/coreComponents/dataRepository/BufferOps_inline.hpp

```diff
@@ -784,6 +784,60 @@ localIndex Unpack( buffer_unit_type const * & buffer,
   return sizeOfUnpackedChars;
 }
 
+template< bool DO_PACKING >
+localIndex Pack( buffer_unit_type * & buffer,
+                 SortedArrayView< localIndex const > const & var,
+                 arrayView1d< localIndex const > const & packList,
+                 arraySlice1d< globalIndex const > const & localToGlobal )
+{
+
+  localIndex length = 0;
+  array1d< localIndex > temp( var.size() );
```

Comment:
I'd get rid of `temp` and just do
```c++
for( auto a : packList )
{ length += var.count( a ); }

for( localIndex a=0; a<length; ++a )
  {
    if( var.count( packList[ a ] ) )
    {
      sizeOfPackedChars += Pack< DO_PACKING >( buffer, localToGlobal[packList[a]] );
    }
  }
```

Reply:
Just curious though, why are we only packing things that are both in `var` and `packList`?

Reply:
Yeah I'm curious about this too, it seems like we also have packing/unpacking versions currently in main that use this logic as well.

Reply:
so, in this specific case, I think I am using this to pack a set. This set (`var`) contains the indices of the fractured elements. I think that the `packList` contains all ghost indices and I only want to pack the ghosts that are fracture and those contained in the set (`var`).

Reply:
It is used here https://github.com/GEOSX/GEOSX/blob/404cf125360352eab85b22f68823166e3f298800/src/coreComponents/mesh/CellElementSubRegion.cpp#L212

Reply:
Cool, thanks! Just seemed a little strange but that makes perfect sense!